### PR TITLE
feat: enable vpc endpoints for each construct

### PR DESCRIPTION
**Pull request by erdemayyildiz**
Fixes #

Issues: https://github.com/awslabs/generative-ai-cdk-constructs/issues/153

Enable Vpc endpoints to each construct to enable communication over private subnet of Vpc.
This change will only enable the VPC settings for the ones constructs create. Customer provided VPC will be used as provided.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Comment by krokoko** [comment]
Thanks @ErdemAyyildiz ! Overall LGTM, could you please list all the samples you deployed with your changes to validate it was working correctly ?
Also, I just saw that the build was failing, could you please have a look and I'll put my review in once it is done

**Comment by ErdemAyyildiz** [comment]
hi @krokoko 

sure, i have fixed the build issue. the sample packages that i tested the code are;

- Document Explorer
- Open Search and Aurora Default Vector Store
- Aurora Construct
- Bedrock Agent

**Review by scottschreckengaust — DISMISSED** [review]

File: src/patterns/gen-ai/aws-web-crawler/index.ts [hunk 1]
```diff
@@ -353,7 +363,7 @@ export class WebCrawler extends BaseClass {
       replaceComputeEnvironment: true,
       updateTimeout: cdk.Duration.minutes(30),
       updateToLatestImageVersion: true,
-      vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS },
+      vpcSubnets: { subnetType: ec2.SubnetType.PRIVATE_ISOLATED },
```
Comment (scottschreckengaust) [inline src/patterns/gen-ai/aws-web-crawler/index.ts:366]: Can you confirm that the crawler would still be able to access web pages with `PRIVATE_ISOLATED`?

**Comment by krokoko** [comment]
LGTM, we will merge after the failing build is fixed :)

**Review by krokoko — APPROVED** [review]

**Review by scottschreckengaust — APPROVED** [review]

**Pull request merged**